webapp-django/crashstats/home/static/home/js/home.js

```diff
@@ -0,0 +1,462 @@
+            var subjectString = this.toString();
+            if (typeof position !== 'number' || !isFinite(position) || Math.floor(position) !== position || position > subjectString.length) {
+                position = subjectString.length;
+            }
+            position -= searchString.length;
+            var lastIndex = subjectString.indexOf(searchString, position);
+            return lastIndex !== -1 && lastIndex === position;
+        };
+    }
+
+    var COLORS = ['#6a3d9a', '#e31a1c', '#008800', '#1f78b4'];
+
+    var container = $('#homepage-graph-container');
+    var superSearchApiUrl = container.data('supersearch-api');
+    var productBuildTypesApiUrl = container.data('product-build-types-api');
+    var adiApiUrl = container.data('adi-api');
+
+    var product = container.data('product');
+    var versions = container.data('versions');
+    var platforms = container.data('platforms');
+    var duration = container.data('duration');
+
+    var pageTitle = $('title').text();
+
+    // Versions ending in -b are "magic" versions that will be split into all
+    // of the beta versions starting with that same version number.
+    var betaVersions = [];
+    versions.forEach(function (version) {
+        if (version.endsWith('b')) {
+            betaVersions.push(version);
+        }
+    });
+
+    // Create date variables. They will be assigned a value in `updateDates()`.
+    var endDate;
+    var startDate;
+
+    /**
+     * Return a string with only the date part of a Date object, in ISO format.
+     */
+    function dateIsoFormat(date) {
+        return date.toISOString().substring(0, 10);
+    }
+
+    /**
+     * History change handler. Reloads the graph's data.
+     */
+    window.onpopstate = function (e) {
+        duration = e.state.duration;
+
+        $('#duration .selected').removeClass('selected');
+        $('#duration .days-' + duration).addClass('selected');
+
+        updateDates();
+        loadData();
+    };
+
+    /**
+     * Click event handler for the duration menu. Changes the history and
+     * reloads the graph's data.
+     */
+    $('#duration').on('click', 'a', function (e) {
+        e.preventDefault();
+
+        var link = $(this);
+
+        // Do nothing if this is already selected.
+        if (link.hasClass('selected')) {
+            return;
+        }
+
+        duration = link.data('duration');
+        var url = link.attr('href');
+
+        $('#duration .selected').removeClass('selected');
+        link.addClass('selected');
+
+        window.history.pushState({ duration: duration }, pageTitle, url);
+
+        updateDates();
+        loadData();
+    });
+
+    /**
+     * Change the dates of the request depending on the duration.
+     */
+    function updateDates() {
+        endDate = new Date();
+        endDate.setDate(endDate.getDate() - 1);  // We start with yesterday.
+
+        startDate = new Date();
+        startDate.setDate(endDate.getDate() - duration);
+    }
```

Comment:
I suspect it's a leapyear bug https://twitter.com/peterbe/status/704641995147444224